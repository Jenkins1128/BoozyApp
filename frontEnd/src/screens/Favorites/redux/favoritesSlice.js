import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
