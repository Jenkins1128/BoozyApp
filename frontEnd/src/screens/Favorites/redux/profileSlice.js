import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
