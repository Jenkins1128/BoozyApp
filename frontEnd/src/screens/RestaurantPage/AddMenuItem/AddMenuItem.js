import React from 'react';
import { View, Text, TextInput, TouchableOpacity, TouchableWithoutFeedback, Animated, StyleSheet, Keyboard } from 'react-native';

const AddMenuItem = ({ bounceValue, menuItemState, dispatch, updatePrice, updateDescription, getDataFromMenuItem, dismiss }) => {
	return (
		<TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
			<Animated.View testID={'subView'} style={[styles.subView, { transform: [{ translateY: bounceValue }] }]}>
				<View testID={'addMenuItemHeader'} style={styles.addMenuItemHeader}>
					<View testID={'addMenuItemTitleContainer'} style={styles.addMenuItemTitleContainer}>
						<Text testID={'addMenuItemTitle'} style={styles.addMenuItemTitle}>
							Add Menu Item
						</Text>
					</View>
					<View>
						<TouchableOpacity testID={'cancelButton'} onPress={dismiss} style={styles.cancelButton}>
							<Text testID={'cancelText'} style={styles.cancelText}>
								Cancel
							</Text>
						</TouchableOpacity>
					</View>
				</View>
				<TextInput
					testID={'priceInput'}
					style={styles.priceStyle}
					keyboardType={'numeric'}
					multiline={false}
					value=''
					returnKeyType='next'
					onChangeText={(price) => dispatch(updatePrice({ price: price }))}
					value={menuItemState.price > 0 ? menuItemState.price.toString() : ''}
					placeholder='Enter Price'
				/>
				<TextInput
					testID={'descriptionInput'}
					style={styles.descriptionStyle}
					multiline={false}
					returnKeyType='next'
					onChangeText={(description) => dispatch(updateDescription({ description: description }))}
					value={menuItemState.description}
					placeholder='Enter menu item'
				/>
				<TouchableOpacity testID={'addMenuItemButton'} onPress={getDataFromMenuItem} style={styles.addMenuItemButton}>
					<Text testID={'addMenuItemText'} style={styles.addMenuItemText}>
						Submit
					</Text>
				</TouchableOpacity>
			</Animated.View>
		</TouchableWithoutFeedback>
	);
};

export default AddMenuItem;

const styles = StyleSheet.create({
	subView: {
		position: 'absolute',
		backgroundColor: '#FFFFFF',
		padding: 20,
		borderRadius: 30
	},
	addMenuItemHeader: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginBottom: 20
	},
	addMenuItemTitleContainer: {
		marginRight: 20
	},
	addMenuItemTitle: {
		fontSize: 20,
		fontWeight: 'bold',
		color: '#EB8873'
	},
	cancelButton: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#FFFFFF',
		padding: 10,
		borderRadius: 20
	},
	priceStyle: {
		fontSize: 18,
		textAlign: 'center',
		marginBottom: 20
	},
	descriptionStyle: {
		fontSize: 18,
		textAlign: 'center',
		marginBottom: 20
	},
	addMenuItemButton: {
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873',
		padding: 10,
		borderRadius: 20
	},
	cancelText: {
		color: '#EB8873',
		fontWeight: 'bold'
	},
	addMenuItemText: {
		color: 'white',
		fontWeight: 'bold'
	}
});
