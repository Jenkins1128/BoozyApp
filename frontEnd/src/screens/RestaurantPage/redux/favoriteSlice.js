import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const favoriteAsync = createAsyncThunk('favoriteAsync/status', async (data, { rejectWithValue }) => {
	const { restaurantId, name } = data;
	try {
		//favorite restaurant
		await axios.post(`${baseUrl}/user/add`, {
			restaurantId: restaurantId,
			name: name
		});
		//add restaurants to your favorites
		const response = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const favoriteSlice = createSlice({
	name: 'favoriteAsync',
	initialState,
	reducers: {
		resetFavorite: (state) => {
			state.currentState.favoriteRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(favoriteAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					favoriteColor: payload[0]['contains'] ? 'red' : 'white',
					favoriteRequestStatus: 'fulfilled'
				});
			})
			.addCase(favoriteAsync.rejected, (state) => {
				state.currentState.favoriteRequestStatus = 'rejected';
			});
	}
});

export const { resetFavorite } = favoriteSlice.actions;

export const selectFavoriteState = (state) => state.favorite.currentState;
export default favoriteSlice.reducer;
