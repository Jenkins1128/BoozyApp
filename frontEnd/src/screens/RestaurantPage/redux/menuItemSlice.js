import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const postMenuItemAsync = createAsyncThunk('postMenuItemAsync/status', async (data, { rejectWithValue }) => {
	const { description, price, restaurantName, restaurantId } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/menu', {
			content: description,
			price: price,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const postMenuItemSlice = createSlice({
	name: 'postMenuItemAsync',
	initialState,
	reducers: {
		updatePrice: (state, { payload }) => {
			state.currentState = { ...state.currentState, price: payload.price };
		},
		updateDescription: (state, { payload }) => {
			state.currentState = { ...state.currentState, description: payload.description };
		},
		resetMenuItem: (state) => {
			state.currentState = { ...state.currentState, price: '', description: '', postMenuItemRequestStatus: 'idle' };
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(postMenuItemAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'fulfilled' };
			})
			.addCase(postMenuItemAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'rejected' };
			});
	}
});

export const { updatePrice, updateDescription, resetMenuItem } = postMenuItemSlice.actions;

export const selectMenuItemState = (state) => state.menuItem.currentState;
export default postMenuItemSlice.reducer;
