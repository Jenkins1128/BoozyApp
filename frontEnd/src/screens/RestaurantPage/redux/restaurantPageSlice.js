import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		restaurantId: '',
		restaurantName: '',
		restaurantImage: 'https://boozyimage.s3.us-east-2.amazonaws.com/logo.png',
		phoneNumber: '',
		allCategories: '',
		rating: 0.0,
		reviews: 0,
		favoriteColor: 'white',
		changeFavoritedColor: '',
		starCount: 0,
		favorited: false,
		rated: false,
		price: 0.0,
		description: '',
		menuItemArray: []
	}
};

export const getMenuItemsAsync = createAsyncThunk('getMenuItemsAsync/status', async (restaurantId, { rejectWithValue }) => {
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/menu/list', {
			restaurantId: restaurantId
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const restaurantPageSlice = createSlice({
	name: 'getMenuItemsAsync',
	initialState,
	reducers: {
		updateState: (state, { payload }) => {
			console.log('payload', payload);
			state.currentState = {
				...state.currentState,
				restaurantId: payload.params.id,
				restaurantName: payload.params.name,
				restaurantImage: payload.params.img,
				phoneNumber: payload.params.phone,
				allCategories: payload.params.allCategories,
				rating: payload.params.rating,
				reviews: payload.params.reviews,
				rated: payload.params.alreadyRated,
				favorited: payload.params.alreadyFavorited,
				starCount: 0
			};
		},
		updateFavoriteColor: (state, { payload }) => {
			state.currentState = {
				...state.currentState,
				favoriteColor: payload.color
			};
		},
		updateStarCount: (state, { payload }) => {
			state.currentState = {
				...state.currentState,
				starCount: payload.starCount
			};
		},
		updateMenuItems: (state, { payload }) => {
			state.currentState = { ...state.currentState, menuItemArray: [...state.currentState.menuItemArray, payload.menuItem] };
		},
		resetRestaurantPageRequestStatus: (state) => {
			state.currentState.restaurantPageRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getMenuItemsAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, restaurantPageRequestStatus: 'fulfilled', menuItemArray: payload };
			})
			.addCase(getMenuItemsAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, restaurantPageRequestStatus: 'rejected' };
			});
	}
});

export const { updateState, updateFavoriteColor, updateStarCount, updateMenuItems, resetRestaurantPageRequestStatus } = restaurantPageSlice.actions;

export const selectRestaurantPageState = (state) => state.restaurantPage.currentState;
export default restaurantPageSlice.reducer;
