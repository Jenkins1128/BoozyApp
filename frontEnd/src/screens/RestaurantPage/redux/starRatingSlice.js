import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const starRatingAsync = createAsyncThunk('starRatingAsync/status', async (data, { rejectWithValue }) => {
	const { rating, restaurantId, restaurantName } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/restaurants', {
			content: rating,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const starRatingSlice = createSlice({
	name: 'starRatingAsync',
	initialState,
	reducers: {
		resetStarRatingRequestStatus: (state) => {
			state.currentState.starRatingRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(starRatingAsync.fulfilled, (state) => {
				state.currentState.starRatingRequestStatus = 'fulfilled';
			})
			.addCase(starRatingAsync.rejected, (state) => {
				state.currentState.starRatingRequestStatus = 'rejected';
			});
	}
});

export const { resetStarRatingRequestStatus } = starRatingSlice.actions;

export const selectStarRatingState = (state) => state.starRating.currentState;
export default starRatingSlice.reducer;
