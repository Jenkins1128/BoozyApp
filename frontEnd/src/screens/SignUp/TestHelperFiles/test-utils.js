import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
