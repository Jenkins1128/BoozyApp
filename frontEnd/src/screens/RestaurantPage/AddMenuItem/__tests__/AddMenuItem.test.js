import React from 'react';
import { render, cleanup, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';
import AddMenuItem from '../AddMenuItem';

describe('<AddMenuItem />', () => {
	let menuItemState;
	let dispatch, updatePrice, updateDescription, getDataFromMenuItem, dismiss;
	let rendered;

	beforeEach(() => {
		menuItemState = {
			price: 6,
			description: 'test'
		};
		(dispatch = jest.fn()), (updatePrice = jest.fn()), (updateDescription = jest.fn()), (getDataFromMenuItem = jest.fn()), (dismiss = jest.fn());
		rendered = render(<AddMenuItem menuItemState={menuItemState} dispatch={dispatch} updatePrice={updatePrice} updateDescription={updateDescription} getDataFromMenuItem={getDataFromMenuItem} dismiss={dismiss} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should render subView styles correctly', () => {
		const subViewComponent = rendered.getByTestId('subView');
		const given = subViewComponent.props.style;
		const result = {
			position: 'absolute',
			backgroundColor: '#FFFFFF',
			padding: 20,
			borderRadius: 30,
			transform: [{ translateY: undefined }]
		};
		expect(result).toMatchObject(given);
	});

	it('should render addMenuItemHeader styles correctly', () => {
		const addMenuItemHeaderComponent = rendered.getByTestId('addMenuItemHeader');
		const given = addMenuItemHeaderComponent.props.style;
		const result = {
			flexDirection: 'row',
			justifyContent: 'space-between',
			alignItems: 'center',
			marginBottom: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render addMenuItemTitleContainer styles correctly', () => {
		const addMenuItemTitleContainerComponent = rendered.getByTestId('addMenuItemTitleContainer');
		const given = addMenuItemTitleContainerComponent.props.style;
		const result = {
			marginRight: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render addMenuItemTitle styles correctly', () => {
		const addMenuItemTitleComponent = rendered.getByTestId('addMenuItemTitle');
		const given = addMenuItemTitleComponent.props.style;
		const result = {
			fontSize: 20,
			fontWeight: 'bold',
			color: '#EB8873'
		};
		expect(result).toMatchObject(given);
	});

	it('should render cancelButton styles correctly', () => {
		const cancelButtonComponent = rendered.getByTestId('cancelButton');
		const given = cancelButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#FFFFFF',
			padding: 10,
			borderRadius: 20,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render priceInput styles correctly', () => {
		const priceInputComponent = rendered.getByTestId('priceInput');
		const given = priceInputComponent.props.style;
		const result = {
			fontSize: 18,
			textAlign: 'center',
			marginBottom: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render descriptionInput styles correctly', () => {
		const descriptionInputComponent = rendered.getByTestId('descriptionInput');
		const given = descriptionInputComponent.props.style;
		const result = {
			fontSize: 18,
			textAlign: 'center',
			marginBottom: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render addMenuItemButton styles correctly', () => {
		const addMenuItemButtonComponent = rendered.getByTestId('addMenuItemButton');
		const given = addMenuItemButtonComponent.props.style;
		const result = {
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			padding: 10,
			borderRadius: 20,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render cancelText styles correctly', () => {
		const cancelTextComponent = rendered.getByTestId('cancelText');
		const given = cancelTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render addMenuItemText styles correctly', () => {
		const addMenuItemTextComponent = rendered.getByTestId('addMenuItemText');
		const given = addMenuItemTextComponent.props.style;
		const result = {
			color: 'white',
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire updatePrice onChangeText event', () => {
		const inputComponent = rendered.getByTestId('priceInput');
		fireEvent(inputComponent, 'changeText', 5);
		expect(updatePrice).toHaveBeenCalledWith({ price: 5 });
	});

	it('should fire updateDescription onChangeText event', () => {
		const inputComponent = rendered.getByTestId('descriptionInput');
		fireEvent(inputComponent, 'changeText', 'pizza, yum');
		expect(updateDescription).toHaveBeenCalledWith({ description: 'pizza, yum' });
	});

	it('should fire addMenuItemButton event', () => {
		const addMenuItemButton = rendered.getByText('Submit');
		fireEvent.press(addMenuItemButton);
		expect(getDataFromMenuItem).toHaveBeenCalledTimes(1);
	});
});
