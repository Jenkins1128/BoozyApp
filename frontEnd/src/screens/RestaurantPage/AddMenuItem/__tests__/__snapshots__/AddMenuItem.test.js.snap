// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddMenuItem /> should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 30,
      "padding": 20,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": undefined,
        },
      ],
    }
  }
  testID="subView"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
      }
    }
    testID="addMenuItemHeader"
  >
    <View
      style={
        Object {
          "marginRight": 20,
        }
      }
      testID="addMenuItemTitleContainer"
    >
      <Text
        style={
          Object {
            "color": "#EB8873",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
        testID="addMenuItemTitle"
      >
        Add Menu Item
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 20,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="cancelButton"
      >
        <Text
          style={
            Object {
              "color": "#EB8873",
              "fontWeight": "bold",
            }
          }
          testID="cancelText"
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    multiline={false}
    onChangeText={[Function]}
    placeholder="Enter Price"
    rejectResponderTermination={true}
    returnKeyType="next"
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
    testID="priceInput"
    underlineColorAndroid="transparent"
    value="6"
  />
  <TextInput
    allowFontScaling={true}
    multiline={false}
    onChangeText={[Function]}
    placeholder="Enter menu item"
    rejectResponderTermination={true}
    returnKeyType="next"
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
    testID="descriptionInput"
    underlineColorAndroid="transparent"
    value="test"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EB8873",
        "borderRadius": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="addMenuItemButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
      testID="addMenuItemText"
    >
      Submit
    </Text>
  </View>
</View>
`;
